prov/sockets/src/sock_ep_msg.c

```diff
@@ -614,74 +634,62 @@ static int sock_ep_cm_connect(struct fid_ep *ep, const void *addr,
 	memcpy(&req->src_addr, _ep->attr->src_addr, sizeof(req->src_addr));
 	memcpy(&handle->dest_addr, addr, sizeof(handle->dest_addr));
 
+	_ep->attr->info.handle = (void*) handle;
 	handle->ep = _ep;
 	handle->req = req;
+	handle->cm_head = &_ep->attr->domain->cm_head;
```

Comment:
We can get to cm_head via handle->ep.  No need to duplicate the path.

Reply:
Done.